src/test/pmemset_source/pmemset_source.c

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+/*
+ * pmemset_source.c -- pmemset_source unittests
+ */
+#include "fault_injection.h"
+#include "libpmemset.h"
+#include "unittest.h"
+#include "ut_pmemset_utils.h"
+#include "out.h"
+#include "source.h"
+
+/*
+ * test_set_from_pmem2_valid - test valid pmemset_source allocation
+ */
+static int
+test_set_from_pmem2_valid(const struct test_case *tc, int argc, char *argv[])
+{
+	if (argc < 1)
+		UT_FATAL("usage: test_set_from_pmem2_valid <file>");
+
+	char *file = argv[0];
+
+	struct pmem2_source *src_pmem2;
+	struct pmemset_source *src_set;
+
+	int fd = OPEN(file, O_RDWR);
+
+	int ret = pmem2_source_from_fd(&src_pmem2, fd);
+	UT_ASSERTeq(ret, 0);
+
+	ret = pmemset_source_from_pmem2(&src_set, src_pmem2);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+	UT_ASSERTne(src_set, NULL);
+
+	UT_ASSERTeq(src_set->type, PMEMSET_SOURCE_PMEM2);
```

Comment:
I know we did these types of tests for libpmem2. But I think that was a mistake.

We should never test internal details of an implementation.